### PR TITLE
WebsiteProfile & Co

### PR DESCRIPTION
**Pull request by romo-odoo**

### CODE REVIEW

File: addons/website_profile/static/src/interactions/website_profile.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        ".send_validation_email": { "t-on-click.prevent": this.onClickSend },
+        ".validated_email_close": { "t-on-click": () => rpc('/profile/validate_email/close') },
+    };
+
+    onClickSend(ev) {
+        const element = ev.currentTarget;
+        rpc('/profile/send_validation_email', {
+            redirect_url: element.dataset["redirect_url"],
+        }).then(function (data) {
+            if (data) {
```
Comment (robinlej) [inline addons/website_profile/static/src/interactions/website_profile.js]: Considering what's inside the `then` is a redirect, why not `const data = await this.waitFor(rpc)`?

File: addons/website_profile/static/src/interactions/website_profile.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import { registry } from "@web/core/registry";
+
+import { redirect } from "@web/core/utils/urls";
+import { rpc } from "@web/core/network/rpc";
+
+export class WebsiteProfile extends Interaction {
+    static selector = ".o_wprofile_email_validation_container";
+    dynamicContent = {
+        ".send_validation_email": { "t-on-click.prevent": this.onClickSend },
+        ".validated_email_close": { "t-on-click": () => rpc('/profile/validate_email/close') },
```
Comment (robinlej) [inline addons/website_profile/static/src/interactions/website_profile.js]: Looks like a good candidate for async + `blockedUntilDone` (+ double quotes `"`)

File: addons/website_profile/static/src/interactions/website_profile_editor.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import { loadWysiwygFromTextarea } from "@web_editor/js/frontend/loadWysiwygFromTextarea";
+
+export class WebsiteProfileEditor extends Interaction {
+    static selector = ".o_wprofile_editor_form"
+    dynamicContent = {
+        '.o_forum_file_upload': { "t-on-change": this.onUploadFile },
+        '.o_forum_profile_pic_edit': { "t-on-click.prevent": this.onClickEditProfilePic },
+        '.o_forum_profile_pic_clear': { "t-on-click": this.onClickClearProfilePic },
+        '.o_forum_profile_bio_edit': { "t-on-click.prevent": this.onClickEditProfileBio },
+        '.o_forum_profile_bio_cancel_edit': { "t-on-click.prevent": this.onClickCancelEditProfileBio },
```
Comment (robinlej) [inline addons/website_profile/static/src/interactions/website_profile_editor.js]: `"`

File: addons/website_profile/static/src/interactions/website_profile_editor.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
+
+import { Interaction } from "@web/public/interaction";
+import { registry } from "@web/core/registry";
+
+import { loadWysiwygFromTextarea } from "@web_editor/js/frontend/loadWysiwygFromTextarea";
+
+export class WebsiteProfileEditor extends Interaction {
+    static selector = ".o_wprofile_editor_form"
```
Comment (robinlej) [inline addons/website_profile/static/src/interactions/website_profile_editor.js]: `;`

File: addons/website_profile/static/src/interactions/website_profile_editor.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    dynamicContent = {
+        '.o_forum_file_upload': { "t-on-change": this.onUploadFile },
+        '.o_forum_profile_pic_edit': { "t-on-click.prevent": this.onClickEditProfilePic },
+        '.o_forum_profile_pic_clear': { "t-on-click": this.onClickClearProfilePic },
+        '.o_forum_profile_bio_edit': { "t-on-click.prevent": this.onClickEditProfileBio },
+        '.o_forum_profile_bio_cancel_edit': { "t-on-click.prevent": this.onClickCancelEditProfileBio },
+    }
```
Comment (robinlej) [inline addons/website_profile/static/src/interactions/website_profile_editor.js]: `;`

File: addons/website_profile/static/src/interactions/website_profile_editor.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+    setup() {
+        this.textareaEl = this.el.querySelector("textarea.o_wysiwyg_loader");
+
+        this.options = {
+            recordInfo: {
+                context: this.context,
```
Comment (robinlej) [inline addons/website_profile/static/src/interactions/website_profile_editor.js]: Nope -> `this.services.website_page.context`

File: addons/website_profile/static/src/interactions/website_profile_editor.js [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    onClickClearProfilePic(ev) {
+        const formEl = ev.currentTarget.closest("form");
+        formEl.querySelector(".o_wforum_avatar_img").src = "/web/static/img/placeholder.png";
+        const inputElement = document.createElement("input");
+        inputElement.setAttribute("name", "clear_image");
+        inputElement.setAttribute("id", "forum_clear_image");
+        inputElement.setAttribute("type", "hidden");
+        formEl.append(inputElement);
```
Comment (robinlej) [inline addons/website_profile/static/src/interactions/website_profile_editor.js]: `this.insert(...)`

File: addons/website_profile/static/src/interactions/website_profile.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    };
+
+    onClickSend(ev) {
+        const element = ev.currentTarget;
+        rpc('/profile/send_validation_email', {
```
Comment (ged-odoo) [inline addons/website_profile/static/src/interactions/website_profile.js]: onClickSend could be async.
```js
const data = await this.waitFor(rpc(...));
if (data) {
  redirect(...);
}
```

File: addons/website_profile/static/src/interactions/website_profile_editor.js [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+    onClickEditProfilePic(ev) {
+        ev.currentTarget.closest("form").querySelector(".o_forum_file_upload").click();
+    }
+
+    onClickEditProfileBio(ev) {
+        ev.currentTarget.classList.add("d-none");
+        document.querySelector(".o_forum_profile_bio_cancel_edit").classList.remove("d-none");
+        document.querySelector(".o_forum_profile_bio_form").classList.remove("d-none");
+        document.querySelector(".o_forum_profile_bio").classList.add("d-none");
```
Comment (robinlej) [inline addons/website_profile/static/src/interactions/website_profile_editor.js]: for both `onClickEditProfileBio` and `onClickCancelEditProfileBio`: you could just set a boolean (directly inlined, no need for a handler) and `t-att-class` the 4 elements

File: addons/website_profile/static/src/interactions/website_profile_next_rank_card.js [hunk 10]
```diff
@@ -0,0 +1,15 @@
+
+import { Interaction } from "@web/public/interaction";
+import { registry } from "@web/core/registry";
+
+class websiteProfileNextRankCard extends Interaction {
```
Comment (ged-odoo) [inline addons/website_profile/static/src/interactions/website_profile_next_rank_card.js]: `WebsiteProfileNextRankCard

File: addons/website_profile/static/src/interactions/website_profile_editor.js [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+        document.querySelector(".o_forum_profile_bio_form").classList.add("d-none");
+        document.querySelector(".o_forum_profile_bio").classList.remove("d-none");
+    }
+
+    onUploadFile(ev) {
+        if (!ev.currentTarget.files.length) {
+            return;
+        }
+        const formEl = ev.currentTarget.closest("form");
+        var reader = new window.FileReader();
```
Comment (robinlej) [inline addons/website_profile/static/src/interactions/website_profile_editor.js]: ~`var`~

File: addons/website_profile/static/src/interactions/website_profile_editor.js [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+        reader.readAsDataURL(ev.currentTarget.files[0]);
+        reader.onload = function (ev) {
```
Comment (robinlej) [inline addons/website_profile/static/src/interactions/website_profile_editor.js]: `this.addListener(reader, "load", ...)`

File: addons/website_profile/static/src/interactions/website_profile_next_rank_card.js [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+    static selector = ".o_wprofile_progress_circle"
```
Comment (robinlej) [inline addons/website_profile/static/src/interactions/website_profile_next_rank_card.js]: `;`

File: addons/website_profile/static/src/interactions/website_profile_next_rank_card.js [hunk 14]
```diff
@@ -0,0 +1,15 @@
+
```
Comment (robinlej) [inline addons/website_profile/static/src/interactions/website_profile_next_rank_card.js]:
```diff
- 
```

File: addons/website_profile/static/src/interactions/website_profile_next_rank_card.js [hunk 15]
```diff
@@ -0,0 +1,15 @@
...
+import { Interaction } from "@web/public/interaction";
+import { registry } from "@web/core/registry";
+
+class websiteProfileNextRankCard extends Interaction {
+    static selector = ".o_wprofile_progress_circle"
+
+    setup() {
+        new Tooltip(this.el.querySelector('g[data-bs-toggle="tooltip"]'));
```
Comment (robinlej) [inline addons/website_profile/static/src/interactions/website_profile_next_rank_card.js]: shouldn't it be `window.Tooltip.getOrCreateInstance(element);` ?

(and then `registerCleanup => bsTooltip.dispose()`)

File: addons/website_profile/static/src/interactions/website_profile.js (hunk 2 above)
Comment (romo-odoo) [reply addons/website_profile/static/src/interactions/website_profile.js]: I don't think this is necessary. Here the function will only we used once + I'm not sure we want it to be async.

File: addons/website_profile/static/src/interactions/website_profile.js [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+    dynamicContent = {
+        ".send_validation_email": { "t-on-click.prevent": this.onClickSend },
+        ".validated_email_close": { "t-on-click": () => rpc("/profile/validate_email/close") },
+    };
+
+    async onClickSend(ev) {
+        const element = ev.currentTarget;
+        const data = await this.waitFor(rpc('/profile/send_validation_email', {
+            redirect_url: element.dataset["redirect_url"],
+        }))
```
Comment (robinlej) [inline addons/website_profile/static/src/interactions/website_profile.js]: `;`

**Pull request merged**